nep-X1.mediawiki

```diff
@@ -0,0 +1,43 @@
+<pre>
+  NEP: 31
+  Title: Contract Destroy Function
+  Author: Owen Zhang <zhangzhihong@ngd.neo.org>
+  Type: Standard
+  Status: Draft
+  Created: 2024-09-11
+</pre>
+
+==Abstract==
+
+This proposal standardizes the destroy function that can be implemented by contracts to have some code executed right after initial deployment or update.
+
+==Motivation==
+
+N3 already provided contract destroy mechanism but it hasn't been summarized in a standard. This standard will show details.
+
+==Specification==
+
+Neo N3 has a native <code>ContractManagement</code> contract that is used for contract destroy via <code>Destroy</code> method. It can delete all storage references under the contract.
+
+===Methods===
+
+====destroy====
+
+<pre>
+{
+  "name": "destroy",
+  "safe": false,
+  "parameters": [],
+  "returntype": "Void"
+}
+</pre>
+
+This method SHOULD invoke <code>Destroy</code> method of <code>ContractManagement</code> contract when a contract is destroyed. After execution, all related storage and contract state will be deleted.  <code>Destroy</code>notification will be emitted by <code>ContractManagement</code> contract in destroy transaction.
```

Comment:
I think it makes sense to describe the `Destroy` notification format (similar to how NEP17 describes the Transfer event